File: 2038.cpp
struct Solution {
    bool winnerOfGame(const string &colors) {
        size_t asum = 0, bsum = 0, lastIdx = 0, n = colors.size();
        char lastChar = colors[0];
        for (size_t i = 1; i < colors.size(); ++i){
            if (colors[i] != lastChar){
                if (i >= 3 + lastIdx){
                    if (lastChar == 'A'){
                        asum += i - lastIdx - 2;
                    } else {
                        bsum += i - lastIdx - 2;
                    }
                }
                lastChar = colors[i];
                lastIdx = i;
            }
        }
        if (n >= 3 + lastIdx){
            if (lastChar == 'A'){
                asum += n - lastIdx - 2;
            } else {
                bsum += n - lastIdx - 2;
            }
        }
        return asum > bsum;
    }
};

File: 2038.py
class Solution:
    def winnerOfGame(self, colors: str) -> bool:
        asum = 0
        bsum = 0
        lastLetter = colors[0]
        lastIdx = 0
        for i, c in enumerate(colors[1:]):
            if c != lastLetter:
                if i - lastIdx >= 2:
                    if lastLetter == 'A':
                        asum += i - lastIdx - 1
                    else:
                        bsum += i - lastIdx - 1
                lastLetter = c
                lastIdx = i + 1
        if len(colors) - lastIdx >= 3:
            if lastLetter == 'A':
                asum += len(colors) - lastIdx - 2
            else:
                bsum += len(colors) - lastIdx - 2
        return asum > bsum

